### ISSUE TITLE
Teach PETScWrappers::TimeStepper that a problem is linear.

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#17526]
@stefanozampini FYI

In step-86, I believe that what PETSc TS really does is solve a nonlinear system in each time step. Of course, the problem is linear, and so the second iteration instantly succeeds at least if we solve linear systems sufficiently accurately. But this is unnecessary: What we should really do is tell PETSc TS that the problem is linear.

We currently have no way of doing that, but I was told this week at the deal.II workshop that PETSc has functions that can be used to inform it that the problem is linear. We should expose this functionality in our wrappers (e.g., by adding a flag to the `AdditionalData` for `PETScWrappers::TimerStepper`).

**Comment by bangerth** [comment]
This can be done with the `TSSetProblemType()` function: https://petsc.org/release/manualpages/TS/TSSetProblemType/

**Comment by fernandezerbes** [comment]
I looked at this issue and I was thinking about adding a parameter to `PETScWrappers::TimeStepperData::add_parameters`

https://github.com/dealii/dealii/blob/3ec1812ce24fe72a05b4691680e411a7bd5f539d/source/lac/petsc_ts.cc#L29-L38

and then use this parameter to decide what type of problem to set with `TSSetProblemType()` in `TimeStepper::reinit`

https://github.com/dealii/dealii/blob/3ec1812ce24fe72a05b4691680e411a7bd5f539d/include/deal.II/lac/petsc_ts.templates.h#L316

Does that sound like a reasonable approach @bangerth?

**Comment by stefanozampini** [comment]
adding an extra entry to the parameters sounds right to me

**Comment by bangerth** [comment]
I agree, that's the right place! Thank you for being willing to work on this!
